common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -200,6 +202,46 @@ def get(request):
     return request.session.get('partial_pipeline')
 
 
+def active_provider_requires_data_sharing(request):
+    """
+    Determine two things - first, whether there's an active third-party
+    identity provider currently running, and second, if that active provider
+    requires data sharing consent in order to proceed.
+    """
+    running_pipeline = get(request)
+    if running_pipeline:
+        current_provider = provider.Registry.get_from_pipeline(running_pipeline)
+        return current_provider and current_provider.require_data_sharing_consent
+    return False
+
+
+def active_provider_requests_data_sharing(request):
+    """
+    Determines two things - first, whether there's an active third-party
+    identity provider currently running, and second, if that active provider
+    requests data sharing consent in order to proceed.
+    """
+    running_pipeline = get(request)
+    if running_pipeline:
+        current_provider = provider.Registry.get_from_pipeline(running_pipeline)
+        return current_provider and current_provider.request_data_sharing_consent
+    return False
+
+
+def get_real_social_auth_object(request):
+    """
+    At times, the pipeline will have a "social" kwarg that contains a dictionary
+    rather than an actual DB-backed UserSocialAuth object. We need the real thing,
+    so this method allows us to get that by passing in the relevant request.
+    """
+    running_pipeline = get(request)
+    if running_pipeline and 'social' in running_pipeline['kwargs']:
+        social = running_pipeline['kwargs']['social']
+        if isinstance(social, dict):
+            social = UserSocialAuth.objects.get(uid=social.get('uid', ''))
```

Comment:
Why is the UID defaulting to an empty string here? Will the empty string result in `social` being None or an empty queryset? Just want to verify the intended behavior here.